## Orthology search and paralogy pruning
[OrthoFinder](https://github.com/davidemms/OrthoFinder) was used to group the extracted coding regions into orthogroups. We used [DIAMOND](https://github.com/bbuchfink/diamond) to BLAST the proteins, [MAFFT](https://github.com/GSLBiotech/mafft) to align the orthogroups and [RAxML](https://github.com/stamatak/standard-RAxML) to infer the genetrees.

    python3 /home/saabalde/bin/OrthoFinder_source/orthofinder.py -f PROTEINS_directory -S diamond -M msa -A mafft -T raxml -t 4 -a 4

OrthoFinder assigned 229,299 genes (92.5% of the total) to 15758 orthogroups. Fifty percent of all genes were in orthogroups with 27 or more genes (G50 was 27) and were contained in the largest 1834 orthogroups (O50 was 1834). There were 2 orthogroups with all species present and 0 of these consisted entirely of single-copy genes.

Since all orthogroups are multicopy, we pruned the paralogs (homolog genes originated from duplication events) using [PhyloPyPruner](https://pypi.org/project/phylopypruner/). But first, we need to prepare the data. Instead of working with all orthogroups, we first filtered out all with fewer than four species: 7,824 orthogroups. We created a list of genes, called "00-Orthogroups.Atleastfour.txt", from the "Orthogroups.GeneCount.tsv" files generated by OrthoFinder. Using that list, we fetched the corresponding alignments and genetrees.

    mkdir -p 01-OrthoFinder_Alignments 02-Orthofinder_Genetrees
    
    while IFS='' read -r LINE || [ -n "${LINE}" ]
        do
        cp 00-OrthoFinder_300aa_final/OrthoFinder/Results_Mar13/MultipleSequenceAlignments/${LINE}.fa ./01-OrthoFinder_Alignments/
    done < 00-Orthogroups.Atleastfour.txt 
    while IFS='' read -r LINE || [ -n "${LINE}" ]
        do
        cp 00-OrthoFinder_300aa_final/OrthoFinder/Results_Mar13/Gene_Trees/${LINE}_tree.txt ./02-Orthofinder_Genetrees
    done < 00-Orthogroups.Atleastfour.txt 

Now, we format the files for PhyloPyPruner.

    # Alignments 
    orthofinder2phylopypruner Species.list 01-OrthoFinder_Alignments

    # Genetrees
    for i in 02-Orthofinder_Genetrees/*tree.txt
        do
        sed -i 's/19_228\_/19_228\./g' $i
        sed -i 's/19_375\_/19_375\./g' $i
        sed -i 's/19_380\_/19_380\./g' $i
        sed -i 's/20_005\_/20_005\./g' $i
        sed -i 's/20_022\_/20_022\./g' $i
        sed -i 's/20_023\_/20_023\./g' $i
        sed -i 's/20_026\_/20_026\./g' $i
        sed -i 's/20_038\_/20_038\./g' $i
        sed -i 's/20_053\_/20_053\./g' $i
        sed -i 's/20_063\_/20_063\./g' $i
        sed -i 's/20_073\_/20_073\./g' $i
        sed -i 's/20_078\_/20_078\./g' $i
        sed -i 's/20_082\_/20_082\./g' $i
        sed -i 's/20_093\_/20_093\./g' $i
        sed -i 's/20_115\_/20_115\./g' $i
        sed -i 's/20_132\_/20_132\./g' $i
        sed -i 's/DRR151142\_/DRR151142\./g' $i
        sed -i 's/P15761_101\_/P15761_101\./g' $i
        sed -i 's/P15761_102\_/P15761_102\./g' $i
        sed -i 's/P15761_110\_/P15761_110\./g' $i
        sed -i 's/P15761_117\_/P15761_117\./g' $i
        sed -i 's/P15761_125\_/P15761_125\./g' $i
        sed -i 's/P15761_133\_/P15761_133\./g' $i
        sed -i 's/P15761_141\_/P15761_141\./g' $i
        sed -i 's/P15761_149\_/P15761_149\./g' $i
        sed -i 's/P15761_157\_/P15761_157\./g' $i
        sed -i 's/P15761_165\_/P15761_165\./g' $i
        sed -i 's/P15761_173\_/P15761_173\./g' $i
        sed -i 's/P15761_181\_/P15761_181\./g' $i
        sed -i 's/P15761_189\_/P15761_189\./g' $i
        sed -i 's/SRR2500940\_/SRR2500940\./g' $i
        sed -i 's/SRR2681155\_/SRR2681155\./g' $i
        sed -i 's/SRR2681679\_/SRR2681679\./g' $i
        sed -i 's/SRR2682004\_/SRR2682004\./g' $i
        sed -i 's/SRR2682099\_/SRR2682099\./g' $i
        sed -i 's/SRR2682154\_/SRR2682154\./g' $i
        sed -i 's/SRR3105702\_/SRR3105702\./g' $i
        sed -i 's/SRR3105703\_/SRR3105703\./g' $i
        sed -i 's/SRR3105704\_/SRR3105704\./g' $i
        sed -i 's/SRR3105705\_/SRR3105705\./g' $i
        sed -i 's/SRR5760179\_/SRR5760179\./g' $i
        sed -i 's/SRR6374833\_/SRR6374833\./g' $i
        sed -i 's/SRR6375633\_/SRR6375633\./g' $i
        sed -i 's/SRR8454219\_/SRR8454219\./g' $i
        sed -i 's/SRR8506641\_/SRR8506641\./g' $i
        sed -i 's/SRR8524599\_/SRR8524599\./g' $i
        sed -i 's/SRR8617822\_/SRR8617822\./g' $i
        sed -i 's/SRR8641368\_/SRR8641368\./g' $i

        sed -i 's/\_OF\_/\.OF\@/g' $i
	
        mv -- "$i" "${i%_tree.txt}.tre"
    done

Now we should be ready to run PhyloPyPruner, but if there is any error in the files format if will let you know

    # Alignments and genetrees need to be in the same directory
    mkdir 03-PhyloPyPruner_input
    cp 01-OrthoFinder_Alignments/* 03-PhyloPyPruner_input/
    cp 02-Orthofinder_Genetrees/*  03-PhyloPyPruner_input/

    # Run PhyloPyPruner
    phylopypruner --dir 03-PhyloPyPruner_input/ --output 04-PhyloPyPruner_output --no-supermatrix --trim-lb 5 --min-support 60 --mask pdist \
                  --outgroup SRR2500940_Xenoturbella_profunda --prune LS --min-taxa 5

If you are not familiar with PhyloPyPruner you should check its very well-documented Wiki, but this is a short description of the selected parameters:
<ul>
    <li>--no-supermatrix: do not concatenate the output into a supermatrix. We will still clean these alignments, so we are not ready for a supermatrix, yet. Besides, this way PhyloPyPruner will be faster.</li>
    <li>--trim-lb: remove long branches, which are those longer than X times the standard deviation of all branch lengths. Here, we followed the tutorial and chose a factor of five.</li>
    <li>--min-support: collapse nodes with a support value below a given threshold (here, Bootstrap < 60) into polytomies.</li>
    <li>--mask: if two sequences from the same OTU are monophyletic, select only one of them. Here, we kept the sequence with the shortest distance to its sister taxa.</li>
    <li>--outgroup: define the outgroup species to root the trees. If Xenoturbella missing, the method used is midpoint root.</li>
    <li>--prune: there are three methods in PhyloPyPruner to prune out paralogs. In this case, I will use LS (Largest Subtree), which searches the tree, from the root to the tips, for subtrees that contain non-repetetive OTUs. The subtree which contains the highest number of sequences is the "largest subtree" and is retained as an ortholog.</li>
    <li>--min-taxa: discard output alignments with fewer than X OTUs. In orthogroups with only four taxa is very easy to have only three unique sequences, which precludes gene tree inference with bootstrap support. Hence, I will select orthogroups containing at least five species.</li>
</ul>

After pruning all paralogs, the number of orthogroups went down to 6,033. The different stats concerning sequences per alignment have also been affected, but that makes sense because now the maximum number of sequences per alignment is set to 48 (the number of species). The percentage of missing data has gone up from 82.60% to 84.90%. Finally, 2781 branches have been removed after being deemed too long, just 1.38% of the total.

## Prepare the data for phylogenomic analysis
We first did some housekeeping in our set of genes. They were: (1) converted to non-aligned fasta files (removing the gaps), (2) stop codons were removed, (3) the headers were cleaned, to leave only the species name, and (4) we corrected the species names that changed during the course of the study.

    mkdir 05-PhyloPyPruner_output-Clean_genes
    for i in 04-PhyloPyPruner_output/phylopypruner_output/output_alignments/*fa
        do
        sed -E '/>/ s/\.OF\@.+//g' $i > ${i}.rename; mv ${i}.rename $i
        sed -i '/>/! s/\-//g' $i
        sed -i 's/\*//g' $i
	
        sed -i 's/19_228.Sterreria_psammicola/19_228.Sterreria_sp/g' $i
        sed -i 's/20_053.Archaphanostoma_agile/20_053.Baltalimania_agile/g' $i
        sed -i 's/P15761_101.Archaphanostoma_occulta/P15761_101.Baltalimania_occulta/g' $i
        sed -i 's/P15761_102.Isodiametra_sp18/P15761_102.Praeconvoluta_tigrina/g' $i
        sed -i 's/P15761_125.Haplopostia_rubropunctata/P15761_125.Thalassoanaperus_rubellus/g' $i
        sed -i 's/P15761_133.Mecynostomum_sp14/P15761_133.Praeaphanostoma_sp/g' $i
        sed -i 's/P15761_173.Sterreria_sp1/P15761_173.Sterreria_rubra/g' $i
        sed -i 's/P15761_181.Sterreria_sp2/P15761_181.Sterreria_lundini/g' $i
        sed -i 's/SRR6374833.Isodiametra_pulchra/SRR6374833.Aphanostoma_pulchra/g' $i
        sed -i 's/SRR8524599.Pseudaphanostoma_variabilis/SRR8524599.Baltalimania_agile/g' $i
	
        mv -- "$i" "${i%.fa}.fasta"
     done

Thus far, this dataset contains species with more than one specimen. We need to remove the duplicated individuals, leaving only one tip per species. We removed the specimen with the fewest number of orthologs:
    <li>20_038.Philactinoposthia_rhammifera</li>
    <li>20_053.Baltalimania_agile</li>
    <li>P15761_149.Solenofilomorpha_sp2</li>
    <li>P15761_165.Kuma_sp5</li>
    <li>P15761_189.Solenofilomorpha_sp9</li>
    <li>SRR8506641.Symsagittifera_roscoffensis</li>

    mkdir 06-No_duplicates
    cp 05-PhyloPyPruner_output-Clean_genes/ 06-No_duplicates/

    for i in 06-No_duplicates/*fasta
        do
        tr '\n' ' ' < $i | sed 's/\ >/\,>/g' | tr ',' '\n' | egrep -v '20_038|20_053|P15761_149|P15761_165|P15761_189|SRR8506641' | tr ' ' '\n' > ${i}.tmp
        mv ${i}.tmp ${i}
    done

    # Remove alignments with fewer than five sequences
    for i in 06-No_duplicates/*fasta
        do
        sequences=$( grep -c '>' $i )
        if [ $sequences -lt 5 ]; then
            rm $i
        fi
    done

One problem of working with transcriptomes is that sometimes it is difficult to resolve the correct assembly, returning artefactual contigs like chimaeras (i.e. contigs that result from merging two genes into one). This can result in problems during orthology inferences or errors in the alignments. To avoid that, we ran [PREQUAL](https://github.com/simonwhelan/prequal), which pairwise compares the sequences of each fasta file to find regions that do not seem to be homologous to the other sequences. These stretches are removed or masked.

    mkdir 07-Prequal
    cp 06-No_duplicates/*fasta 07-Prequal/

    for i in *fasta
        do
        prequal $i > $i.log
        rm $i
    done

PREQUAL has removed a lot of sequences that were generally bad, as well as masked regions within other sequences that were kept. However, some of these masked regions are larger than we would liked, so we decided to remove all sequences that do not have, at least, 250 unmasked amino acids.

    mkdir 08-Prequal_filter_masked
    cp 07-Prequal/*filtered 08-Prequal_filter_masked

    cd 08-Prequal_filter_masked/

    # Remove masked regions
    for i in *filtered
        do
        sed -i '/>/! s/X//g' $i
    done

To remove sequences shorter than 250 amino acids, we used the "removesmalls.pl" perl script shared in this [Biostars post](https://www.biostars.org/p/79202/).

    for i in *filtered
        do
        perl ./removesmalls.pl 250 $i > $i.fasta
        rm $i
    done

    # Remove fasta files with fewer than five sequences
    for i in *fasta
        do
        seqs=$( grep -c '>' $i)
        if [ $seqs -lt 5 ]; then
            rm $i
        fi
    done

Now, we can compare the original and the edited files to remove all sequences that do not have at least 250 unmasked amino acids:

    # Create a list of the remaining fasta files
    ls *fasta | sed 's/.filtered.fasta//g' > fasta_files.txt

    # Copy the corresponding Prequal files into this directory
    while IFS='' read -r LINE || [ -n "${LINE}" ]
        do
        cp ../../04-NoDuplicates/${LINE} ./${LINE}.filtered
    done < ./fasta_files.txt


ls *fasta | sed 's/.filtered.fasta//g' > fasta_files.txt

while IFS='' read -r LINE || [ -n "${LINE}" ]
    do
    cp ../../04-NoDuplicates/${LINE} ./${LINE}.filtered
done < ./fasta_files.txt 

for i in $( ls *filtered | sed 's/.filtered//g' )
    do
    grep '>' ${i}.filtered.fasta | sort > header_fasta; grep '>' ${i}.filtered | sort > header_filtered
    fgrep -x -f header_fasta  header_filtered > Similarities.txt
    sed -i 's/>//g' Similarities.txt
    
	while read sequence_id
        do
        bioawk -v var="${sequence_id}" -c fastx '$name ~ var {print ">"$name"\n"$seq}' $i.filtered >> $i.filtered.rename
    done < ./Similarities.txt
	
done
rm *fasta *filtered *txt

for i in *rename
    do
	mv -- "$i" "${i%.fasta.filtered.rename}.fasta.filtered"
done

## This step has removed 1748 orthogroups, from 5569 to 3821, more than a 30% of the fasta files. We can perform the alignments now, but for
## that we need to prepare two datasets: one with and one without Notocelis. The dataset with Notocelis will only include those orthogroups
## where that species is present.























## Later on, I will prepare to parallel datasets: with and without the species Notocelis gullmarensis. This species has proven to be a problematic branch in the tree. Previous phylogentic attempts (not shown) have recovered Notocelis in different positions of the tree, lowering the support of the nodes where it was positioned.
