## Orthology search and paralogy pruning
[OrthoFinder](https://github.com/davidemms/OrthoFinder) was used to group the extracted coding regions into orthogroups. We used [DIAMOND](https://github.com/bbuchfink/diamond) to BLAST the proteins, [MAFFT](https://github.com/GSLBiotech/mafft) to align the orthogroups and [RAxML](https://github.com/stamatak/standard-RAxML) to infer the genetrees.

    python3 /home/saabalde/bin/OrthoFinder_source/orthofinder.py -f PROTEINS_directory -S diamond -M msa -A mafft -T raxml -t 4 -a 4

OrthoFinder assigned 229,299 genes (92.5% of the total) to 15758 orthogroups. Fifty percent of all genes were in orthogroups with 27 or more genes (G50 was 27) and were contained in the largest 1834 orthogroups (O50 was 1834). There were 2 orthogroups with all species present and 0 of these consisted entirely of single-copy genes.

Since all orthogroups are multicopy, we pruned the paralogs (homolog genes originated from duplication events) using [PhyloPyPruner](https://pypi.org/project/phylopypruner/). But first, we need to prepare the data. Instead of working with all orthogroups, we first filtered out all with fewer than four species: 7,824 orthogroups. We created a list of genes, called "00-Orthogroups.Atleastfour.txt", from the "Orthogroups.GeneCount.tsv" files generated by OrthoFinder.

## Now that we have the orthogroups, it is a matter of filtering out potential paralogs, contaminants, gene sets that do not represent real
## orthogroups and so on. To do so, I will apply a series of filters that aim to recover a reliable gene set. Hopefully.
## I will start by selecting all orthogroups with at least four species present. I will do that manually, sorting the table produced by
## OrthoFinder, and copying the corresponding alignments and gene trees to a new directory: 7824 orthogroups.
